src/falcon/falcon_c/rpo.h

```diff
@@ -0,0 +1,36 @@
+#include <stdint.h>
+#include <string.h>
+
+/* ==================================================================== */
+/*
+ * RPO hashing algorithm related structs and methods.
+ */
+
+typedef struct
+{
+    union
+    {
+        uint64_t A[12];
+        uint8_t dbuf[96];
+    } st;
+    uint64_t dptr;
+} inner_rpo128_context;
+
+void rpo128_inc_init(inner_rpo128_context *rc);
+void rpo128_inc_absorb(inner_rpo128_context *rc, const uint8_t *in, size_t len);
+void rpo128_inc_finalize(inner_rpo128_context *rc);
+void rpo128_inc_squeeze(inner_rpo128_context *rc, uint8_t *out, size_t len);
+
+#define inner_rpo128_init(rc) rpo128_inc_init(rc)
+#define inner_rpo128_inject(rc, in, len) rpo128_inc_absorb(rc, in, len)
+#define inner_rpo128_flip(rc) rpo128_inc_finalize(rc)
+#define inner_rpo128_extract(rc, out, len) rpo128_inc_squeeze(rc, out, len)
```

Comment:
Do we need to re-define the function names? Since these are going to be called only from our own code, maybe we can just use the original names? (but I'm fine with either).

Reply:
Makes sense. Removed.